### PR TITLE
feat(b8): first implementation

### PR DESCRIPTION
**Pull request by rokam**
First implementation of B8 type of device (vaccum cleaner).

## Summary by CodeRabbit

- **New Features**
  - Introduced support for Midea B8 devices, including robust message processing and attribute management.
  - Added detailed enums for various statuses, modes, and error types related to Midea B8 devices.
  - Enhanced device interaction capabilities with specific methods for queries, message processing, and attribute settings.

- **Tests**
  - Implemented a comprehensive suite of unit tests for validating the functionality of the Midea B8 device.

### CODE REVIEW

**Comment by rokam** [comment]
@coderabbitai review

**Comment by tufeikafei** [comment]
cool, thank you very much.
I will test it first.

**Comment by rokam** [comment]
> cool, thank you very much. I will test it first.

There is still some job to be done.

File: midealocal/devices/b8/message.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+class MessageSet(MessageB8Base):
+    """B8 message set."""
+
+    def __init__(self, protocol_version: int) -> None:
+        """Initialize B8 message set."""
+        super().__init__(
+            protocol_version=protocol_version,
+            message_type=MessageType.set,
+            body_type=0x22,
```
Comment (chemelli74) [inline midealocal/devices/b8/message.py:79]: Does a const makes sense here ?

File: midealocal/devices/b8/message.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+class MessageSetCommand(MessageB8Base):
+    """B8 message set command."""
+
+    def __init__(self, protocol_version: int, work_mode: B8WorkMode) -> None:
+        """Initialize B8 message set command."""
+        super().__init__(
+            protocol_version=protocol_version,
+            message_type=MessageType.set,
+            body_type=0x22,
```
Comment (chemelli74) [inline midealocal/devices/b8/message.py:111]: Here as well

**Pull request merged**